### PR TITLE
 Disable colorized git output in GitProxy#branch

### LINKED ISSUE #4056: local bundle config throws error about branch, even though gemfile specifies correct branch
Bundler version 1.10.6
git version 2.5.3
ruby 2.2.3 (via RVM)

A couple of my projects have a dependency for another gem that I have cloned locally for development.

In my Gemfile I have:

<pre>
gem 'my_gem', :git=>'git://github.com/path_to/my_gem.git', :branch=>'rails4'
</pre>

To override that locally, I use:
`bundle config local.my_gem /Path/to/gem`

Here's the error, though:
<pre>
$ bundle show my_gem
Local override for my_gem at /Users/stephenmarsh/dev/my_gem is using branch rails4 but Gemfile specifies rails4
</pre>

Obviously this is a very confusing error because the branches are identical/apparently correct.

I have tried combinations of: re-pulling that branch from origin, running `bundle update my_gem`, and deleting and re-cloning the my_gem repo from scratch.  None of these work.

### PR DESCRIPTION
**Pull request by agis-**
Fixes #4056.

### CODE REVIEW

File: lib/bundler/source/git/git_proxy.rb [hunk 1]
```diff
@@ -64,7 +64,7 @@ def revision
 
         def branch
           @branch ||= allowed_in_path do
-            git("branch") =~ /^\* (.*)$/ && $1.strip
+            git("branch --no-color") =~ /^\* (.*)$/ && $1.strip
```
Comment (segiddins) [inline lib/bundler/source/git/git_proxy.rb]: Isn't there a `rev-parse` command we could use here?

Comment (segiddins) [inline lib/bundler/source/git/git_proxy.rb]: Maybe `git rev-parse --abbrev-ref HEAD`

File: spec/install/gemfile/git_spec.rb [hunk 2]
```diff
@@ -996,4 +996,27 @@
...
   end
+
+  describe "when the git source is overriden with a local git repo" do
+    before do
+      bundle "config --global local.foo #{lib_path("foo")}"
+    end
+
+    describe "and git output is colorized" do
+      before do
+        File.write("#{ENV["HOME"]}/.gitconfig", "[color]\n\tui = always\n")
```
Comment (segiddins) [inline spec/install/gemfile/git_spec.rb]: `undefined method `write' for File:Class` on 1.8.7

**Comment by agis-** [comment]
@segiddins Updated!

**Comment by indirect** [comment]
Thanks! @homu r+

**Comment by indirect** [comment]
Uhh, hmm... maybe it has to be at the start of a line?

@homu r+

**Comment by indirect** [comment]
Oh cool, Homu's down right now. At least we still have the merge button. :P

**Pull request merged**